model/account.js

```diff
@@ -0,0 +1,115 @@
+/***
+ * cuHacking 2020 - Account
+ * 
+ * This file is responsible for interacting with Firebase Auth to manage user logins/accounts
+ * 
+ */
+
+const Database = require('./database');
+const ROLES = {"public": 0, "user": 1, "admin": 2};
+
+let fbAccount, fbAdmin;
+
+var Account = module.exports;
+
+Account.init = function(firebase, admin){
+    
+    fbAccount = firebase.auth();
+    fbAdmin = admin.auth();
+    fbAccount.setPersistence(firebase.auth.Auth.Persistence.NONE);
+}
+
+Account.create = function(email, password){
+    
+    let promise = new Promise(function(resolve, reject){
+        
+        // TODO: consider any email or password validation we want to do at the backend level
```

Comment:
Questions like these arise when there's a disconnect between client and server. In our case, our client and server are fully managed by us, so you should allow for this validation to stay in the front end.

If, for instance, we were making a _public_ REST API, then we would be concerned about the information we're given.